src/asm/lexer.c

```diff
@@ -2048,6 +2049,85 @@ static int yylex_SKIP_TO_ENDC(void)
 	return skipIfBlock(true);
 }
 
+static int yylex_SKIP_TO_ENDR(void)
+{
+	dbgPrint("Skipping remainder of REPT/FOR block\n");
+	lexer_SetMode(LEXER_NORMAL);
+	int depth = 1;
+	bool atLineStart = lexerState->atLineStart;
+
+	/* Prevent expanding macro args and symbol interpolation in this state */
+	lexerState->disableMacroArgs = true;
+	lexerState->disableInterpolation = true;
+
+	for (;;) {
+		if (atLineStart) {
+			int c;
+
+			for (;;) {
+				c = peek(0);
+				if (!isWhitespace(c))
+					break;
+				shiftChars(1);
+			}
+
+			if (startsIdentifier(c)) {
+				shiftChars(1);
+				switch (readIdentifier(c)) {
+				case T_POP_FOR:
+				case T_POP_REPT:
+					depth++;
+					break;
+
+				case T_POP_ENDR:
+					depth--;
+					if (!depth)
+						goto finish;
```

Comment:
Shouldn't this be `assert`ed not to happen? `ENDR` is *not* part of a `REPT`/`FOR`'s body, so this shouldn't be possible at all.

Reply:
That's for nested loops which it skips past to find its own matching `ENDR`.

Reply:
Agreed for `depth--;`, but I was talking about the `!depth` check. This should never happen, should it?

Reply:
`depth` starts at 1 since it's already inside a `REPT`/`FOR`. It reaches 0 at the matching `ENDR`, so it returns 0, which makes `yylex` call `yywrap` and end the loop. (Just calling `yywrap` in `fstk_Break` wasn't working.)